gr-digital/python/digital/qa_scrambler.py

```diff
@@ -44,24 +48,64 @@ def tearDown(self):
         self.tb = None
 
     def test_scrambler_descrambler(self):
-        src_data = (1,)*1000
+        src_data = np.random.randint(0,2,500,dtype=np.int8)
+        src = blocks.vector_source_b(src_data, False)
+        scrambler = digital.scrambler_bb(0x85, 1, 6)     # p(x) = x^7 + x^2 + 1
+        dst_tap = blocks.vector_sink_b()
+        descrambler = digital.descrambler_bb(0x85, 0b111, 6) # we can use seed 0 here, it is descrambling.
+        dst = blocks.vector_sink_b()
+        self.tb.connect(src, scrambler, descrambler, dst)
+        self.tb.connect(scrambler, dst_tap)
+        self.tb.run()
+        self.assertEqual(tuple(src_data[:-7]), dst.data()[7:]) # skip garbage during synchronization
+        self.assertEqual(tuple(np.convolve(dst_tap.data(),[1,0,0,0,0,1,0,1])%2)[7:-10],
+                tuple(src_data[:-10])) # manual descrambling test
+
+    def test_scrambler_descrambler_big(self):
+        mask,k = make_mask(51,6,0) #p(x) = x^51+x^6+1
+        src_data = np.random.randint(0,2,1000,dtype=np.int8)
         src = blocks.vector_source_b(src_data, False)
-        scrambler = digital.scrambler_bb(0x8a, 0x7F, 7)     # CCSDS 7-bit scrambler
-        descrambler = digital.descrambler_bb(0x8a, 0x7F, 7)
+        scrambler = digital.scrambler_bb(mask, 1, k)     # p(x) = x^7 + x^2 + 1
```

Comment:
I suggest to leave the old QA code in place and add more tests instead. That'll make it easier to merge this PR.

Reply:
I will comply, but I will add;

The QA has room for even more improvement, it did not correctly detect a bug that was fixed with this commit: https://github.com/gnuradio/gnuradio/commit/fddb0d4ebe36516ee7caa3ebf25cf411b62db6c0#diff-930c609d625aa95d48a093a3e8e06f2ea0e211ba5fb863683bf2342d95574f66

I believe the QA was not updated this time, so it was not improved to detect this error.

I'm pretty sure the QA right now does not actually use the scrambler it intends to. It does complete without error, because it undoes what it does, it does not check that what it does, is correct.